lib/model/CResourceMonitor.cc

```diff
@@ -282,21 +285,26 @@ CResourceMonitor::SResults CResourceMonitor::createMemoryUsageReport(core_t::TTi
 }
 
 std::size_t CResourceMonitor::adjustedUsage(std::size_t usage) const {
-    // Background persist causes the memory size to double due to copying
-    // the models. On top of that, after the persist is done we may not
-    // be able to retrieve that memory back. Thus, we report twice the
-    // memory usage in order to allow for that.
-    // See https://github.com/elastic/x-pack-elasticsearch/issues/1020.
-    // Issue https://github.com/elastic/x-pack-elasticsearch/issues/857
-    // discusses adding an option to perform only foreground persist.
-    // If that gets implemented, we should only double when background
-    // persist is configured.
-
-    // We also scale the reported memory usage by the inverse of the byte limit margin.
+    // We  scale the reported memory usage by the inverse of the byte limit margin.
     // This gives the user a fairer indication of how close the job is to hitting
     // the model memory limit in a concise manner (as the limit is scaled down by
     // the margin during the beginning period of the job's existence).
-    size_t adjustedUsage = static_cast<std::size_t>(2 * usage / m_ByteLimitMargin);
+    size_t adjustedUsage{static_cast<std::size_t>(usage / m_ByteLimitMargin)};
+
+    if (m_PersistenceInForeground == false) {
+        // Background persist causes the memory size to double due to copying
+        // the models. On top of that, after the persist is done we may not
+        // be able to retrieve that memory back. Thus, we report twice the
+        // memory usage in order to allow for that.
+        // See https://github.com/elastic/x-pack-elasticsearch/issues/1020.
+        // Issue https://github.com/elastic/x-pack-elasticsearch/issues/857
+        // discusses adding an option to perform only foreground persist.
+        // If that gets implemented, we should only double when background
+        // persist is configured.
```

Comment:
I'd be inclined to have a function something like `std::size_t persistenceMemoryIncreaseFactor() { return m_PersistenceInForeground ? 1 : 2 }`. Then you don't need to duplicate this comment.